### PR TITLE
Add/format using blocks

### PR DESCRIPTION
**Pull request by ozdemir08**

### CODE REVIEW

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidEventsClient.cs [hunk 1]
```diff
@@ -25,36 +25,37 @@ public void FetchAllEvents(DataSource source, Action<ResponseStatus, List<IEvent
         {
             callback = ToOnGameThread(callback);
             using (var task = mEventsClient.Call<AndroidJavaObject>("load", source == DataSource.ReadNetworkOnly ? true : false))
-            {   // Task<AnnotatedData<EventBuffer>> task
-                task.Call<AndroidJavaObject>("addOnSuccessListener", new TaskOnSuccessProxy<AndroidJavaObject>(
+            {
+                TaskListenerHelper.AddOnSuccessListener<AndroidJavaObject>(
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidEventsClient.cs]: Can we move those methods from TaskListenerHelper to AndroidTaskUtils?

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs [hunk 2]
```diff
@@ -294,21 +293,22 @@ public void FetchAllSavedGames(DataSource source, Action<SavedGameRequestStatus,
...
+                        callback(SavedGameRequestStatus.InternalError, new List<ISavedGameMetadata>()));
             }
         }
 
         public void Delete(ISavedGameMetadata metadata)
         {
             AndroidSnapshotMetadata androidMetadata = metadata as AndroidSnapshotMetadata;
             Misc.CheckNotNull(androidMetadata);
-            mSnapshotsClient.Call<AndroidJavaObject>("delete", androidMetadata.JavaMetadata);
+            using (mSnapshotsClient.Call<AndroidJavaObject>("delete", androidMetadata.JavaMetadata))
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs]: using (mSnapshotsClient.Call<AndroidJavaObject>("delete", androidMetadata.JavaMetadata));

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs [hunk 3]
```diff
@@ -430,23 +430,25 @@ private static AndroidJavaObject AsMetadataChange(SavedGameMetadataUpdate update
                 if (update.IsCoverImageUpdated)
                 {
                     using (var bitmapFactory = new AndroidJavaClass("android.graphics.BitmapFactory"))
+                    using (var bitmap = bitmapFactory.CallStatic<AndroidJavaObject>(
+                        "decodeByteArray", update.UpdatedPngCoverImage, /* offset= */0, update.UpdatedPngCoverImage.Length))
+                    using (builder.Call<AndroidJavaObject>("setCoverImage", bitmap))
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs]: using (builder.Call<AndroidJavaObject>("setCoverImage", bitmap));

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTaskUtils.cs [hunk 4]
```diff
@@ -52,5 +52,29 @@ public void onFailure(AndroidJavaObject exception)
             mCallback(exception);
         }
     }
+
+    class TaskListenerHelper
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTaskUtils.cs]: Rename it to AndroidTaskUtils

**Comment by ozdemir08** [comment]
All empty using blocks have been replaced.
TaskListenerHelper has been replaced with AndroidTaskUtils

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs [hunk 5]
```diff
@@ -98,38 +99,37 @@ public AndroidSavedGameClient(AndroidJavaObject account)
...
+            {
+                AndroidTaskUtils.AddOnSuccessListener<AndroidJavaObject>(
+                    task,
                     dataOrConflict => {
                         if (dataOrConflict.Call<bool>("isConflict"))
                         {
                             using (var conflict = dataOrConflict.Call<AndroidJavaObject>("getConflict"))
                             {
-                                // If we get here, manual conflict resolution is required.
                                 AndroidSnapshotMetadata original = new AndroidSnapshotMetadata(conflict.Call<AndroidJavaObject>("getSnapshot"));
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs]: using for conflict.Call<AndroidJavaObject>("getSnapshot") and for conflict.Call<AndroidJavaObject>("getConflictingSnapshot")

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs [hunk 6]
```diff
@@ -147,24 +147,23 @@ public AndroidSavedGameClient(AndroidJavaObject account)
...
                                 conflictCallback(resolver, original, null, unmerged, null);
                             }
-                        } else {
+                        }
+                        else
+                        {
                             using (var snapshot = dataOrConflict.Call<AndroidJavaObject>("getData"))
                             {
-                                completedCallback(SavedGameRequestStatus.Success, new AndroidSnapshotMetadata(snapshot.Call<AndroidJavaObject>("freeze")));
+                                AndroidJavaObject metadata = snapshot.Call<AndroidJavaObject>("freeze");
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidSavedGameClient.cs]: using for AndroidJavaObject metadata = snapshot.Call<AndroidJavaObject>("freeze");

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTaskUtils.cs [hunk 7]
```diff
@@ -52,5 +52,23 @@ public void onFailure(AndroidJavaObject exception)
             mCallback(exception);
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTaskUtils.cs]: using (exception) {
  mCallback(exception);
}

File: source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTaskUtils.cs [hunk 8]
```diff
@@ -9,7 +9,7 @@ class TaskOnCompleteProxy<T> : AndroidJavaProxy
     {
         private Action<T> mCallback;
 
-        public TaskOnCompleteProxy(Action<T> callback) 
+        public TaskOnCompleteProxy(Action<T> callback)
```
Comment (olehkuznetsov) [inline source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTaskUtils.cs]: We can make those classes internal to AndroidTaskUtils as they shouldn't be used directly
For TaskOnCompleteProxy<AndroidJavaObject> and TaskOnSuccessProxy<AndroidJavaObject> we can try to overload those cases and add using for object around callback.

**Pull request merged**